ui/src/AuthService.js

```diff
@@ -0,0 +1,37 @@
+import config from './config/config';
+import { requestConfig } from './DataService';
+
+/**
+ * Sends a HTTP request
+ *
+ * @param {string} path - The URI to send the request to
+ * @param {*} method - HTTP Method (PUT, GET, PATCH, DELETE, POST)
+ * @param {*} body - The request payload
+ * @returns {boolean|JSON} - Returns the response from the HTTP request
+ */
+const request = async (path, method, body) => {
+  const response = await fetch(`${config.auth.url}/${path}`, requestConfig(method, body));
+  if (!response.ok) {
+    const msg = await response.text();
```

Comment:
I wonder if the `await` is actually needed here, as `response` should be returned from the fetch call at this point - is `.wait()` asynchronous?

Reply:
Yes it is needed. `fetch` is a web API that returns a `Promise`.



The Fetch API provides an interface for fetching resources (including across the network). It will seem familiar to anyone who has used XMLHttpRequest, but the new API provides a more powerful and flexible feature set.
Concepts and usage
Section

Fetch provides a generic definition of Request and Response objects (and other things involved with network requests). This will allow them to be used wherever they are needed in the future, whether it’s for service workers, Cache API and other similar things that handle or modify requests and responses, or any kind of use case that might require you to generate your own responses programmatically.

It also provides a definition for related concepts such as CORS and the HTTP origin header semantics, supplanting their separate definitions elsewhere.

For making a request and fetching a resource, use the WindowOrWorkerGlobalScope.fetch() method. It is implemented in multiple interfaces, specifically Window and WorkerGlobalScope. This makes it available in pretty much any context you might want to fetch resources in.

> The fetch() method takes one mandatory argument, the path to the resource you want to fetch. It returns a Promise that resolves to the Response to that request, whether it is successful or not.

So we can use it like:

```js
fetch(...).then(...).catch(...);
```

Or by using `async/await`.

https://dev.to/johnpaulada/synchronous-fetch-with-asyncawait

Reply:
Resolving. Hadn't been sure that `.text()` returns a promise, but [it does](https://developer.mozilla.org/en-US/docs/Web/API/Body/text) so await required.